app/services/auto_assignment/assignment_service.rb

```diff
@@ -0,0 +1,99 @@
+class AutoAssignment::AssignmentService
+  pattr_initialize [:inbox!]
+
+  def perform_for_conversation(conversation)
+    return false unless assignable?(conversation)
+
+    agent = find_available_agent
+    return false unless agent
+
+    assign_conversation(conversation, agent)
+  end
+
+  def perform_bulk_assignment(limit: 100)
+    return 0 unless inbox.enable_auto_assignment?
+
+    assigned_count = 0
+
+    unassigned_conversations(limit).each do |conversation|
+      assigned_count += 1 if perform_for_conversation(conversation)
+    end
+
+    assigned_count
+  end
+
+  private
+
+  def assignable?(conversation)
+    inbox.enable_auto_assignment? &&
+      conversation.status == 'open' &&
+      conversation.assignee_id.nil?
+  end
+
+  def unassigned_conversations(limit)
+    scope = inbox.conversations.unassigned.open
+
+    # Apply conversation priority from config
+    scope = apply_conversation_priority(scope)
+    scope.limit(limit)
+  end
+
+  def apply_conversation_priority(scope)
+    case assignment_config['conversation_priority']
+    when 'longest_waiting'
+      scope.order(last_activity_at: :asc, created_at: :asc)
+    else
+      scope.order(created_at: :asc)
+    end
+  end
```

Comment:
```suggestion
  def unassigned_conversations(limit)
  scope = inbox.conversations.unassigned.open
  
  if assignment_config['conversation_priority'] == 'longest_waiting'
    scope = scope.order(:last_activity_at, :created_at)
  else
    scope = scope.order(:created_at)
  end
  
  scope.limit(limit)
end
```

Keeping it a single function seems more readable.